src/content/guides/deploying-istio/index.md

```diff
@@ -0,0 +1,283 @@
+
+Another good practice is to name the service ports. Istio uses the name to discover the protocol used by the end service container. Thus you have to prefix the port name with the protocol desired. The different supported protocols (http, http2, grpc, mongo, or redis) leverage Istio to route traffic more intelligently. Otherwise, it will use plain TCP routing.
+```
+apiVersion: v1
+kind: Service
+metadata:
+  name: app
+spec:
+  ports:
+  - name: http-app
+    port: 80
+    protocol: TCP
+    targetPort: 80
+```
+
+In the strange case, you use multiple services laying over the same pods, you have to use the same kind of traffic for the ports defined in the service.
+
+```
+# WRONG because the protocol is different and same port
+apiVersion: v1             apiVersion: v1
+kind: Service              kind: Service
+metadata:                  metadata:
+  name: app-svc-1            name: app-svc-2
+spec:                      spec:
+  ports:                     ports:
+  - name: http-app           - name: http2-app
+    port: 80                   port: 80
+    protocol: TCP              protocol: TCP
+  selector:                  selector:    
+    app: my-app                app: my-app
+```
+
+Finally, it is good to mention how you can integrate Istio within your existing workloads. By default mutual TLS is not configured, mainly for avoiding downtime issues it can provoke. But lately, there has appeared a new option which you can set the TLS policy as `permissive` and let the proxy accept traffic encrypted and unencrypted. Thus safe wise you can start with this policy and slowly move your services to a `strict` mode TLS in case it is required.
+
+## Installation
+
+Regardless of the provider, the first step to install Istio is to create all cluster role definitions. The service mesh uses different Kubernetes resources to manage the variety of entities in which it relies on. For example, it defines the `gateway` resource for control how the mesh is exposed to Internet.
+
+```
+# Post all resources to the Kubernetes API
+$ kubectl apply -f install/kubernetes/helm/istio/templates/crds.yaml
+```
+
+Istio is a complex system composed by several pieces, by default the installation release all components to the cluster. However, the mesh can be created with different flavors. As an example, you can only deploy `Pilot`, which will give you the ability to enforce service policies. Let's enumerate the different main components:
+
+- Ingress / Egress gateways 
+- Sidecar injector webhook
+- Galley
+- Mixer
+- Pilot
+- Grafana
+- Prometheus
+- Tracing (Jaeger) 
+- Kiali
+- Mutual TLS
+- Certmanager
+
+All of those are deployed but mutual TLS and certmanager. You can configure the components tweaking the values passed to helm. Run `helm inspect install/kubernetes/helm/istio` to display the different options available.
```

Comment:
`All of those are deployed but mutual TLS and certmanager.` -> `With exception of Mutual TLS and Certmanager, all of those are installed.`

Question: What's with the two exceptions? Can they be installed manually? Is it recommendable?

Reply:
Mutual TLS is explained in recommendations. Cert manager is like an addon. I can add more info .good point